src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -56,23 +62,55 @@ public DefaultTargetService(ClientConfig clientConfig) {
     this.targetHttpClient.addDefaultHeader(SDK_USER_KEY, SDK_USER_VALUE);
     this.targetHttpClient.addDefaultHeader(SDK_VERSION_KEY, SDK_VERSION);
     this.clientConfig = clientConfig;
+    this.telemetryService = new TelemetryService(clientConfig);
   }
 
   @Override
   public TargetDeliveryResponse executeRequest(TargetDeliveryRequest deliveryRequest) {
+    TimingTool timer = new TimingTool();
+    timer.timeStart(TIMING_EXECUTE_REQUEST);
+
     String url = clientConfig.getUrl(getBestLocationHint(deliveryRequest));
+    TargetDeliveryResponse targetDeliveryResponse;
+    synchronized (this) {
+      if(!storedTelemetries.isEmpty()) {
```

Comment:
we don't need this section to be synchronized and we don't need to check if `storedTelemetries` is empty, we should just poll `storedTelemetries` and insert the polled entries in a new ArrayList, then check if that one is non-empty and use `Telemetry.entries` instead of `Telemetry.addEntriesItem` to set the entries, if any.

Reply:
^ this should also be moved to `TelemetryService.getTelemetry()`